src/editor/dag.rs

```diff
@@ -468,7 +468,11 @@ impl<'arena, Node: Ast<'arena>> DAG<'arena, Node> {
                 let new_node_name = new_node.display_name();
                 // Add the new child to the children of the cloned cursor
                 cloned_parent.insert_child(new_node, this.arena, insert_index)?;
-
+                // move the cursor to the correct location
+                *this.current_cursor_path.last_mut().unwrap() += match side {
+                    Side::Prev => 0,
+                    Side::Next => 1,
+                };
```

Comment:
This code works, but the index of the new node is already stored in `insert_index`, so this statement can be turned into `*this.current_cursor_path.last_mut().unwrap() = insert_index;`. This avoids unnecessary duplication and makes the code easier to read.

Also, we should have a comment explaining why the `unwrap` is safe.

Unrelated note: As a general rule of thumb, if you're coding and you find two pieces of code that look the same (in this case there are two `match side { ... }`s) then it's usually a good idea to check whether or not they are exactly the same and can be combined in some way.  Because if we want to change something in that code then it's all too easy to forget to change everything and accidentally introduce bugs.

Reply:
Cool I will fix ASAP.